BorrowBookControl.java

```diff
@@ -2,116 +2,116 @@
+
+    private List<book> pending;
+    private List<loan> completed;
+    private book book;
+
+
+    public BorrowBookControl() {
+        this.library = library.INSTANCE();
+        this.state = ControlState.INITIALISED;
+    }
+
+
+    public void setUI(BorrowBookUI UI) {
+        if (!this.state.equals(ControlState.INITIALISED))
+            throw new RuntimeException("BorrowBookControl: cannot call setUI except in INITIALISED state");
+
+        this.UI = UI;
+        UI.setState(BorrowBookUI.UIState.READY);
+        this.state = ControlState.READY;
+    }
+
+
+    public void swiped(int memberId) {
+        if (!this.state.equals(ControlState.READY))
+            throw new RuntimeException("BorrowBookControl: cannot call cardSwiped except in READY state");
+
+        this.member = library.MEMBER(memberId);
+        if (this.member == null) {
+            this.UI.display("Invalid memberId");
+            return;
+        }
+        if (this.library.MEMBER_CAN_BORROW(this.member)) {
+            this.pending = new ArrayList<>();
+            this.UI.setState(BorrowBookUI.UIState.SCANNING);
+            this.state = ControlState.SCANNING; }
+        else
+        {
+            this.UI.display("Member cannot borrow at this time");
+            this.UI.setState(BorrowBookUI.UIState.RESTRICTED); }}
+
+
+    public void scanned(int bookId) {
+        this.book = null;
+        if (!this.state.equals(ControlState.SCANNING)) {
+            throw new RuntimeException("BorrowBookControl: cannot call bookScanned except in SCANNING state");
+        }
+        this.book = this.library.Book(bookId);
+        if (this.book == null) {
+            this.UI.display("Invalid bookId");
+            return;
+        }
+        if (!this.book.AVAILABLE()) {
+            this.UI.display("Book cannot be borrowed");
+            return;
+        }
+        this.pending.add(this.book);
+        for (book displayBook : this.pending) {
+            this.UI.display(displayBook.toString());
+        }
+        if (this.library.Loans_Remaining_For_Member(this.member) - this.pending.size() == 0) {
+            this.UI.display("Loan limit reached");
+            this.complete();
+        }
+    }
+
+
+    public void complete() {
+        if (this.pending.size() == 0) {
+            this.cancel();
+        }
+        else {
+            this.UI.display("\nFinal Borrowing List");
+            for (book displayBook : this.pending) {
+                this.UI.display(displayBook.toString());
+            }
+            this.completed = new ArrayList<loan>();
+            this.UI.setState(BorrowBookUI.UIState.FINALISING);
+            this.state = ControlState.FINALISING;
+        }
+    }
+
+
+    public void commitLoans() {
+        if (!this.state.equals(ControlState.FINALISING)) {
+            throw new RuntimeException("BorrowBookControl: cannot call commitLoans except in FINALISING state");
+        }
+        for (book displayBook : this.pending) {

```

Comment:
D9: (reference D6 and D7) temporary variable 'displayBook' must be a noun and must be meaningful

Reply:
updated